Start "Choose a user interface" conceptual guide

davisagli opened the pull request:
Fixes #1353


<!-- readthedocs-preview plone6 start -->
----
📚 Documentation preview 📚: https://plone6--1749.org.readthedocs.build/

<!-- readthedocs-preview plone6 end -->

davisagli committed:
start choose-user-interface chapter

petschki commented:
From the ClassicUI point of view I would mention that the Frontend is a "rewrite" of Barceloneta (Plone 5) with Twitter Bootstrap 5 ... this maybe catches long term plone developers with well known terms ...

davisagli committed:
add note about bootstrap

davisagli commented:
@petschki done

yurj commented:
> From the ClassicUI point of view I would mention that the Frontend is a "rewrite" of Barceloneta (Plone 5) with Twitter Bootstrap 5 ... this maybe catches long term plone developers with well known terms ...

Are you sure the correct name is "Twitter Bootstrap"? I think it is just Bootstrap.

davisagli committed:
Twitter Bootstrap -> Bootstrap

davisagli commented:
@yurj Thanks. I don't see any mention of "Twitter Bootstrap" on their website. Updated.

stevepiercy committed:
Merge branch '6.0' into choose-ui

stevepiercy committed:
Minor tweaks to glossary.md, with a cross-reference from element to blocks.

stevepiercy committed:
Rephrase the first sentence.

stevepiercy committed:
Polish up classic-ui/index.md

stevepiercy committed:
Added some spit and polish around the excellent content that we have needed for over two years. Thank you, and amen, @davisagli!
- Add note about migration between UIs.
- Explain purpose of this document.
- Switch MyST markup from colon to backtick to support colon-based options.
- Eliminated spacing around and between the two columns.
- Set breakpoints to display columns side-by-side on only widest displays, else it gets smooshed up into an illegible mess.
- Added a target to make the images clickable so they can be viewed in all their glory.
- Update Glossary with missing terms and add references.

stevepiercy committed:
One too many spaces on the indent

stevepiercy reviewed (dismissed):
This is freaking AWESOME @davisagli! Thank you so much for leading the charge on this. I've done my review by making commits and pushing them. Back to you for one final review, and I'm OK to merge this.

We'll probably need to work on a few merge conflicts with other open PRs, too.

davisagli committed:
Merge remote-tracking branch 'origin/6.0' into choose-ui

davisagli committed:
fix merge

davisagli committed:
Clarify the distinction in layout capabilities

davisagli reviewed (commented):
@stevepiercy I resolved conflicts and clarified one point based on feedback from @tisto. I think this is ready to go.

stevepiercy commented on docs/conceptual-guides/choose-user-interface.md:
```diff
@@ -0,0 +1,67 @@
+
+-   The user interface is similar to Plone 5.
+-   Editors create a page using a {term}`WYSIWYG` editor, {term}`TinyMCE`.
+-   Additional widgets can be added to predefined locations, using {term}`portlets`. More sophisticated page layout requires the use of add-ons.
```
```suggestion
-   Additional widgets can be added to predefined locations, using {term}`portlets`.
    More sophisticated page layout requires the use of add-ons.
```

stevepiercy commented on docs/conceptual-guides/choose-user-interface.md:
```diff
@@ -0,0 +1,67 @@
+-   The visual design is based on the {term}`Barceloneta` theme from Plone 5, but updated to use Bootstrap 5.
+-   Content is stored as HTML.
+-   Customization of themes is not well-documented.
+
```
I'm not sure if this is worth mentioning.

```suggestion
-   You can migrate to Volto.

```

stevepiercy commented on docs/conceptual-guides/choose-user-interface.md:
```diff
@@ -0,0 +1,67 @@
+-   Python skills are not required, but can be helpful for extending the backend.
+-   Content is stored as structured JSON.
+-   Customization of themes is well-documented and relatively easy for developers who have experience with React.
+
```
Is this and its counterpart worth mentioning?

```suggestion
-   You cannot migrate to Classic UI.

```

davisagli replied:
I was thinking of maybe adding a separate "Migrating between Volto and Classic UI" heading at the bottom of this page that covers this topic, but I think we're already at the point where it makes sense to merge this as is and make incremental improvements later.

stevepiercy reviewed (dismissed):
LGTM. I'll make the one commit for one sentence per line. What do you think about the two suggestions?

stevepiercy committed:
Update docs/conceptual-guides/choose-user-interface.md

stevepiercy reviewed (approved):
I approve. I'll leave the suggestions for you to decide. It's ready to merge at your discretion.

davisagli committed:
Merge branch '6.0' into choose-ui

davisagli commented:
Last test failure is just a read timeout on the link checker. I'll merge.
